crates/isograph_schema/src/magic_mutation_fields.rs

```diff
@@ -496,3 +435,264 @@ fn skip_arguments_contained_in_field_map(
+        }
+    }
+}
+
+impl<'de, T: ToString> MapAccess<'de> for NameValuePairVec<'de, T> {
+    type Error = ProcessTypeDefinitionError;
+
+    fn next_key_seed<K>(&mut self, seed: K) -> Result<Option<K::Value>, Self::Error>
+    where
+        K: de::DeserializeSeed<'de>,
+    {
+        if let Some(name_value_pair) = self.arguments.get(self.field_idx) {
+            return seed
+                .deserialize(NameSerializer { name_value_pair })
+                .map(Some);
+        }
+        Ok(None)
+    }
+
+    fn next_value_seed<V>(&mut self, seed: V) -> Result<V::Value, Self::Error>
+    where
+        V: de::DeserializeSeed<'de>,
+    {
+        match self.arguments.get(self.field_idx) {
+            Some(name_value_pair) => {
+                self.field_idx += 1;
+                seed.deserialize(ValueSerializer { name_value_pair })
+            }
+            _ => Err(ProcessTypeDefinitionError::FailedToDeserialize(format!(
+                "Called deserialization of field value for a field with idx {} that doesn't exist",
+                self.field_idx
+            ))),
+        }
+    }
+}
+
+struct NameSerializer<'a, TName, TValue: ValueType> {
+    name_value_pair: &'a NameValuePair<TName, TValue>,
+}
+
+struct ValueSerializer<'a, TName, TValue: ValueType> {
+    name_value_pair: &'a NameValuePair<TName, TValue>,
+}
+
+impl<'de, TName: ToString, TValue: ValueType> Deserializer<'de>
+    for NameSerializer<'de, TName, TValue>
+{
+    type Error = ProcessTypeDefinitionError;
+
+    fn deserialize_any<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        visitor.visit_string(self.name_value_pair.name.item.to_string())
+    }
+
+    fn deserialize_identifier<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        visitor.visit_string(self.name_value_pair.name.item.to_string())
+    }
+
+    serde::forward_to_deserialize_any! {
+        bool i8 i16 i32 i64 i128 u8 u16 u32 u64 u128 f32 f64 char str string
+        bytes byte_buf option unit unit_struct newtype_struct seq tuple
+        tuple_struct map struct enum ignored_any
+    }
+}
+
+pub struct ConstantValueDeserializer<'de> {
+    value: &'de ConstantValue,
+}
+
+impl<'de> IntoDeserializer<'de, ProcessTypeDefinitionError> for &'de ConstantValue {
+    type Deserializer = ConstantValueDeserializer<'de>;
+
+    fn into_deserializer(self) -> Self::Deserializer {
+        ConstantValueDeserializer { value: self }
+    }
+}
+
+impl<'de> Deserializer<'de> for ConstantValueDeserializer<'de> {
+    type Error = ProcessTypeDefinitionError;
+
+    fn deserialize_any<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        match self.value {
+            ConstantValue::Boolean(bool) => visitor.visit_bool(*bool),
+            ConstantValue::Enum(enum_literal) => visitor.visit_string(enum_literal.to_string()),
```

Comment:
`visit_str` or `visit_borrowed_str(enum_literal.lookup())` will avoid the need to allocate a new String